src/plugins/yawn/objstack.h

```diff
@@ -0,0 +1,525 @@
+/* FILE NAME:   objstack.h
+
+   Copyright (C) 1997-2015 Vladimir Makarov.
+
+   Written by Vladimir Makarov <vmakarov@gcc.gnu.org>
+
+   This is part of package for work with stack of objects; you can
+   redistribute it and/or modify it under the terms of the GNU Library
+   General Public License as published by the Free Software
+   Foundation; either version 2, or (at your option) any later
+   version.
+
+   This software is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Library General Public License for more details.
+
+   You should have received a copy of the GNU Library General Public
```

Comment:
I am afraid we cannot redistribute this here.

Reply:
Where should I put this file then?

Reply:
Isn't this header file present in some package?

If not please ask the author if it is okay to relicense it as BSD.

Reply:
Downloading the file causes other problems (in the build server...).

Why isn't this file present in a yaep dev package?

Did you get a response from the author?

Reply:
> Downloading the file causes other problems (in the build server...).

I do not think that is a problem. If the CMake code is unable to download the file, then it will just remove the plugin.

> Why isn't this file present in a yaep dev package?

I assume, because you can use another function that offers the same functionality (such as `malloc`) instead. That is only an assumption though.

> Did you get a response from the author?

I did not ask him about relicensing the source code.

Reply:
> I do not think that is a problem. If the CMake code is unable to download the file, then it will just remove the plugin.

That assumes that the user does not care if the plugin gets added or not.

>  I did not ask him about relicensing the source code.

Please do, I think this is by far the easiest option that does not cause any problems. Iirc no author declined such a request up to now.

Reply:
The latest version of the code uses `malloc` as allocation function for the parsing code. While I assume that is not an ideal solution, the update removes the dependency on `objstack.h`.

Reply:
Why is it not an ideal situation? Sounds okay to me?

If a user wants to avoid malloc, he/she will also wants to avoid it at all other places of Elektra, so elektraMalloc need to be changed anyway.

Reply:
> Why is it not an ideal situation? Sounds okay to me?

I assume the author of YAEP would not have written memory management code, if a simple `malloc` would provide the same performance. Anyway, I really do not care if we use `malloc` or something different, as long as the memory management code is reasonable simple to use.

> If a user wants to avoid malloc, he/she will also wants to avoid it at all other places of Elektra, so elektraMalloc need to be changed anyway.

I replaced `malloc` with `elektraMalloc` in the latest commit.